grpc/src/main/java/io/stargate/grpc/service/interceptors/HeadersInterceptor.java

```diff
@@ -0,0 +1,30 @@
+package io.stargate.grpc.service.interceptors;
+
+import io.grpc.Context;
+import io.grpc.Contexts;
+import io.grpc.Metadata;
+import io.grpc.Metadata.Key;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.stargate.grpc.service.Service;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HeadersInterceptor implements ServerInterceptor {
```

Comment:
I went with a separate component, but this could also be merged with `AuthenticationInterceptor` since it does similar work.